packages/backend-core/src/sql/sql.ts

```diff
@@ -113,34 +114,81 @@ function generateSelectStatement(
   knex: Knex
 ): (string | Knex.Raw)[] | "*" {
   const { resource, meta } = json
+  const client = knex.client.config.client as SqlClient
 
   if (!resource || !resource.fields || resource.fields.length === 0) {
     return "*"
   }
 
-  const schema = meta?.table?.schema
+  const schema = meta.table.schema
   return resource.fields.map(field => {
-    const fieldNames = field.split(/\./g)
-    const tableName = fieldNames[0]
-    const columnName = fieldNames[1]
-    const columnSchema = schema?.[columnName]
-    if (columnSchema && knex.client.config.client === SqlClient.POSTGRES) {
-      const externalType = schema[columnName].externalType
-      if (externalType?.includes("money")) {
-        return knex.raw(
-          `"${tableName}"."${columnName}"::money::numeric as "${field}"`
-        )
-      }
+    const parts = field.split(/\./g)
+    let table: string | undefined = undefined
+    let column: string | undefined = undefined
+
+    // Just a column name, e.g.: "column"
+    if (parts.length === 1) {
+      column = parts[0]
+    }
+
+    // A table name and a column name, e.g.: "table.column"
+    if (parts.length === 2) {
+      table = parts[0]
+      column = parts[1]
+    }
+
+    // A link doc, e.g.: "table.doc1.fieldName"
+    if (parts.length > 2) {
+      table = parts[0]
+      column = parts.slice(1).join(".")
```

Comment:
This is honestly something I don't fully understand, but when querying against link tables you get gives fields like `a.doc1.fieldName`, which splits to `["a", "doc1", "fieldName"]`. Previously, with `"${field} as ${field}"` as the return value, the `"fieldName"` segment is preserved despite not being used below, as `columnName = fieldNames[1]` would result in `columnName = "doc1"`, which I don't actually think appears in any table's schema.

However, _not_ preserving this `"fieldName"` segment results in weird errors in other tests. Preserving it fixes them. I did not dig deeply into this.